lib/multi.c

```diff
@@ -1117,11 +1125,16 @@ static CURLMcode Curl_multi_wait(struct Curl_multi *multi,
   nfds += extra_nfds; /* add the externally provided ones */
 
 #ifdef ENABLE_WAKEUP
+#ifdef USE_WINSOCK
+  if(use_wakeup) {
```

Comment:
Maybe check for WSA_INVALID_EVENT?

Reply:
Of course this check is only relevant if my comment below is not the solution.

Reply:
It's no longer possible to get here with that state (since the init function bails if creating the event fails), and it's not specific to the wakeup case anymore either. Could add a `DEBUGASSERT(multi->wsa_event);` at the top of the function if you like?

Reply:
Good idea, thanks!